File: app/src/main/java/com/hillal/hhhhhhh/data/AppDatabase.java
package com.hillal.hhhhhhh.data;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import com.hillal.hhhhhhh.data.model.Transaction;
import com.hillal.hhhhhhh.data.model.Account;
import com.hillal.hhhhhhh.data.model.User;
import com.hillal.hhhhhhh.data.room.TransactionDao;
import com.hillal.hhhhhhh.data.room.AccountDao;
import com.hillal.hhhhhhh.data.room.UserDao;

@Database(entities = {Transaction.class, Account.class, User.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    public abstract TransactionDao transactionDao();
    public abstract AccountDao accountDao();
    public abstract UserDao userDao();
} 

File: android_app/app/src/main/java/com/hsaby/accounting/data/local/AppDatabase.kt
package com.hsaby.accounting.data.local

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.hsaby.accounting.data.local.dao.AccountDao
import com.hsaby.accounting.data.local.dao.TransactionDao
import com.hsaby.accounting.data.local.dao.UserDao
import com.hsaby.accounting.data.local.entity.AccountEntity
import com.hsaby.accounting.data.local.entity.TransactionEntity
import com.hsaby.accounting.data.local.entity.UserEntity

@Database(
    entities = [
        UserEntity::class,
        AccountEntity::class,
        TransactionEntity::class
    ],
    version = 1,
    exportSchema = false
)
@TypeConverters(Converters::class)
abstract class AppDatabase : RoomDatabase() {
    abstract fun userDao(): UserDao
    abstract fun accountDao(): AccountDao
    abstract fun transactionDao(): TransactionDao
} 